nomad/structs/structs_test.go

```diff
@@ -6024,3 +5917,22 @@ func TestTaskGroup_validateScriptChecksInGroupServices(t *testing.T) {
 		require.Nil(t, mErrOK)
 	})
 }
+
+func requireMultierrorContaining(t *testing.T, err error, expected ...string) {
+	t.Helper()
+	require.Error(t, err)
+	mErr, ok := err.(*multierror.Error)
+	require.True(t, ok)
+
+	var found []string
+	for _, e := range expected {
+		for _, actual := range mErr.Errors {
+			if strings.Contains(actual.Error(), e) {
+				found = append(found, e)
+				break
+			}
+		}
+	}
+
+	require.Equal(t, expected, found)
```

Comment:
This would report expected errors and the subset that have been found on failure.  It'd be nice to also report the actual error messages as well.